### PR TITLE
parser.y: fix variable in set, 'set xx.xx = 42' is a valid statement

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

'xx.xx' is a valid variable name in the set statement

Fix https://github.com/pingcap/parser/issues/428

Before:

```
mysql> set SESSION xx.xx = 123;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 15 near ".xx = 123" 
```

After:

```
mysql> set global xx.xx=666;
ERROR 1193 (HY000): Unknown system variable 'xx.xx'
```

### What is changed and how it works?

```
VariableName:
	Identifier
|	Identifier '.' VariableName
	{
		$$ = $1 + "." + $3
	}
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -6436,20 +6437,27 @@ SetExpr:
 EqOrAssignmentEq:
     eq | assignmentEq
 
+VariableName:
+	Identifier
```
Comment (kennytm) [inline parser.y]: Don't this branch needs `$$ = $1`?

File: parser.y [hunk 2]
```diff
@@ -6436,20 +6437,27 @@ SetExpr:
...
+|	Identifier '.' VariableName
```
Comment (wjhuang2016) [inline parser.y]: Should it be `Identifier '.' Identifier`?
```
mysql> set xx.xx.xx=666;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '.xx=666' at line 1
```

File: parser.y (hunk 1 above)
Comment (tiancaiamao) [reply parser.y]: No, the default action for the rule is `$$ = $1`

File: parser.y (hunk 2 above)
Comment (tiancaiamao) [reply parser.y]: Done

**Comment by tiancaiamao** [comment]
PTAL @kennytm @jackysp

**Comment by leoppro** [comment]
LGTM

**Pull request merged**